docs/api.md

```diff
@@ -2738,11 +2738,11 @@ Shortcut for [page.mainFrame().waitFor(selectorOrFunctionOrTimeout[, options[, .
 #### page.waitForFileChooser([options])
 
 - `options` <[WaitTimeoutOptions](####WaitTimeoutOptions)> Optional waiting parameters
-- returns: <[Promise]<[FileChooser]>> A promise that resolves after a page requests a file picker.
+- returns: <[Promise]<[FileChooser]>> A promise that resolves after a page requests a file picker _due to file upload_.
```

Comment:
This is not accurate — the file does not have to be "uploaded" to any server at all.

Reply:
Updated the docs as per you suggested.